src/core/CMakeLists.txt

```diff
@@ -101,6 +123,13 @@ set(libraryname idyntree-core)
 
 add_library(${libraryname} ${IDYNTREE_CORE_EXP_SOURCES} ${IDYNTREE_CORE_EXP_HEADERS} ${IDYNTREE_CORE_EXP_PRIVATE_INCLUDES})
 
+if (DEFINED ${CMAKE_COMPILER_IS_GNUCXX} 
```

Comment:
We can also change strategy: https://github.com/robotology/idyntree/blob/2470a71cee49ee3a611195e36b8dc447eca9f154/cmake/iDynTreeOptions.cmake#L100